staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -139,6 +142,19 @@ func newStore(c *clientv3.Client, codec runtime.Codec, newFunc, newListFunc func
 	w.getCurrentStorageRV = func(ctx context.Context) (uint64, error) {
 		return storage.GetCurrentResourceVersionFromStorage(ctx, s, newListFunc, resourcePrefix, w.objectType)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		ctx := context.Background()
+		err := etcdfeature.DefaultFeatureSupportChecker.CheckClient(ctx, c, storage.RequestWatchProgress)
+		if err != nil {
+			klog.ErrorS(err, "Couldn't check features supported by etcd. Needed to use ConsistentListFromCache")
+		}
```

Comment:
With CheckClient being asynchronous the only case it will return error is a when a invalid feature is passed. In that case would be best to return the error up instead of just logging it and allowing apiserver to continue bootstrap.

Reply:
Done. Add error as return argument of function and modified usages.